### PR TITLE
Update to latest Flutter SDK

### PR DESCRIPTION
**Pull request by rxlabz**

### CODE REVIEW

File: example/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+dependencies:
+  flutter:
+    sdk: flutter
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  flutter_flux:
+    path: /Users/rxlabz/dev/examples/flutter/examples/redux_styles/flutter_flux-1
```
Comment (sethladd) [inline example/pubspec.yaml]: I think we should remove this absolute path, yes?

Comment (rxlabz) [reply example/pubspec.yaml]: it will probably help 😊

File: analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+analyzer:
+  language:
+    enableStrictCallChecks: true
+    enableSuperMixins: true
+    enableAssertInitializer: true
+  strong-mode:
+    implicit-dynamic: false
+
```
Comment (Hixie) [inline analysis_options.yaml]: we shouldn't need this file, `flutter analyze` should automatically get the options from the flutter repo.

File: example/README.md [hunk 3]
```diff
@@ -1,10 +1,7 @@
...
-To run the example, follow the instructions at [flutter.io](https://flutter.io)
-for setting up a flutter environment and connecting your Android or iOS device.
+Mock Chat app example
 
-Then do this:
+```bash
 
-1. `cd chat_app`
-2. `flutter run`
+```
```
Comment (Hixie) [inline example/README.md]: Presumably this was unintentional?

File: example/analysis_options.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+analyzer:
+  language:
+    enableStrictCallChecks: true
+    enableSuperMixins: true
+    enableAssertInitializer: true
+  strong-mode:
+    implicit-dynamic: false
```
Comment (Hixie) [inline example/analysis_options.yaml]: same comment as above

File: example/android/build.gradle [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+rootProject.buildDir = '../build'
+subprojects {
+    project.buildDir = "${rootProject.buildDir}/${project.name}"
+    project.evaluationDependsOn(':app')
+}
+
+task clean(type: Delete) {
+    delete rootProject.buildDir
+}
```
Comment (Hixie) [inline example/android/build.gradle:31]: please don't check in the .jar files or the gradlew and gradlew.bat files

File: example/lib/main.dart [hunk 6]
```diff
@@ -4,8 +4,7 @@
 
 import 'package:flutter/material.dart';
 import 'package:flutter_flux/flutter_flux.dart';
-
-import 'stores.dart';
+import 'package:flutter_flux_example/stores.dart';
```
Comment (Hixie) [inline example/lib/main.dart]: this shouldn't be imported using `package:`

File: example/lib/main.dart [hunk 7]
```diff
@@ -16,6 +15,8 @@ void main() {
 }
 
 class ChatScreen extends StoreWatcher {
+  final TextEditingController msgController = new TextEditingController();
+
   ChatScreen({Key key}) : super(key: key);
```
Comment (Hixie) [inline example/lib/main.dart]: nit: constructor first

see our style guide for more details on flutter style
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

File: example/lib/main.dart [hunk 8]
```diff
@@ -60,9 +62,9 @@ class ChatScreen extends StoreWatcher {
             new AppBar(title: new Text('Chatting as ${chatUserStore.me.name}')),
         body: new Column(children: <Widget>[
           new Flexible(
-              child: new Block(
+              child: new ListView(
                   padding: new EdgeInsets.symmetric(horizontal: 8.0),
-                  scrollAnchor: ViewportAnchor.end,
+                  /*scrollAnchor: ViewportAnchor.end,*/
```
Comment (Hixie) [inline example/lib/main.dart]: don't commit commented out code

File: lib/src/store_watcher.dart [hunk 9]
```diff
@@ -14,17 +14,18 @@
 
 import 'dart:async';
 
-import 'package:meta/meta.dart';
 import 'package:flutter/widgets.dart';
 import 'package:flutter_flux/src/store.dart';
+import 'package:meta/meta.dart';
```
Comment (Hixie) [inline lib/src/store_watcher.dart]: import flutter/foundation.dart instead

File: lib/src/store_watcher.dart [hunk 10]
```diff
@@ -121,9 +126,8 @@ abstract class StoreWatcherMixin implements State<dynamic> { // ignore: TYPE_ARG
     // TODO(abarth): We cancel our subscriptions in [dispose], which means we
     // shouldn't receive this callback when we're not mounted. If that's the
     // case, we should change this check into an assert that we are mounted.
-    if (!mounted)
-      return;
-    setState(() { });
+    if (!mounted) return;
```
Comment (Hixie) [inline lib/src/store_watcher.dart]: this and the other change below seem to be extraneous and violate the flutter style guide.
there's various other things that could be done in this file to improve its adherence to the style guide.

File: lib/src/store_watcher.dart [hunk 11]
```diff
@@ -121,9 +126,8 @@ abstract class StoreWatcherMixin implements State<dynamic> { // ignore: TYPE_ARG
...
+    setState(() {});
```
Comment (Hixie) [inline lib/src/store_watcher.dart]: add a comment inside the closure saying what state changed

File: lib/src/store_watcher.dart [hunk 12]
```diff
@@ -145,8 +149,7 @@ class StoreToken {
 
   @override
   bool operator ==(dynamic other) {
-    if (other is! StoreToken)
-      return false;
+    if (other is! StoreToken) return false;
```
Comment (Hixie) [inline lib/src/store_watcher.dart]: we should really compare runtimeTypes not use `is` here.

**Comment by Hixie** [comment]
Overall this is great, just some minor issues.

File: example/android/build.gradle (hunk 5 above)
Comment (rxlabz) [reply example/android/build.gradle:31]: Sorry, I'm not sure to understand, isn't it the default generated build.gradle ?

File: lib/src/store_watcher.dart (hunk 10 above)
Comment (rxlabz) [reply lib/src/store_watcher.dart]: this 2 changes are only caused by dartfmt. How can I try to improve it , any advice ?

Comment (Hixie) [reply lib/src/store_watcher.dart]: We don't use dartfmt. It doesn't implement Flutter's style guide.

File: example/android/build.gradle (hunk 5 above)
Comment (Hixie) [reply example/android/build.gradle:31]: Sorry I put the comment nearest the files in the diff. It wasn't a comment on build.gradle. Github won't let me comment on the actual binary files.

The patch includes the gradle wrapper jar file and the shell scripts. It should not.

Comment (rxlabz) [reply example/android/build.gradle:31]: Thanks, I removed the gradle dir and gradlew files

File: lib/src/store_watcher.dart (hunk 10 above)
Comment (rxlabz) [reply lib/src/store_watcher.dart]: I didn't know that, I restored the original formatting.
So there is no automated way to format to Flutter's style guide ?

**Comment by rxlabz** [comment]
@hixie Do I need to do something more, or better ?

File: lib/src/store_watcher.dart [hunk 13]
```diff
@@ -85,7 +79,7 @@ class StoreWatcherState extends State<StoreWatcher> with StoreWatcherMixin {
 /// Listens to changes in a number of different stores.
 ///
 /// Used by [StoreWatcher] to track which stores the widget is listening to.
-abstract class StoreWatcherMixin implements State<dynamic> { // ignore: TYPE_ARGUMENT_NOT_MATCHING_BOUNDS, https://github.com/dart-lang/sdk/issues/25232
+abstract class StoreWatcherMixin <T extends StatefulWidget> implements State<T>{
```
Comment (Hixie) [inline lib/src/store_watcher.dart]: nit: remove space before the `<`

File: lib/src/action.dart [hunk 14]
```diff
@@ -58,7 +58,7 @@ class Action<T> implements Function {
     // a [Stream]-based action implementation. At smaller sample sizes this
     // implementation slows down in comparison, yielding average times of 0.1 ms
     // for stream-based actions vs. 0.14 ms for this action implementation.
-    return Future.wait(_listeners
+    return Future.wait<dynamic>(_listeners
         .map((OnData l) => new Future<dynamic>.microtask(() => l(payload))));
   }
```
Comment (Hixie) [inline lib/src/action.dart]: nit: flutter style would be:
```dart
return Future.wait<dynamic>(
  _listeners.map(
    (OnData listener) => new Future<dynamic>.microtask(() => listener(payload)),
  ),
);
￼
```

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
Let me know once you've fixed the two trivial issues I just pointed out and I can land it for you!

Thanks so much for your contribution! And sorry about the delay in responding.

**Comment by rxlabz** [comment]
@hixie Thanks for the review. the two last issues are fixed now !
( so happy to make my first PR to the Flutter repo ( and to a major OSS project ) !  :) )

**Comment by Hixie** [comment]
Hm, I just noticed the cla bot didn't trigger. Let me look into that before I land it.

**Comment by Hixie** [comment]
Oh never mind, it did trigger, we just don't have the labels set.

**Comment by Hixie** [comment]
Ok, all's good! I'm landing it! Thanks so much for the contribution!

**Pull request merged**